pkg/operator/ceph/file/subvolumegroup/controller.go

```diff
@@ -219,6 +219,18 @@ func (r *ReconcileCephFilesystemSubVolumeGroup) reconcile(request reconcile.Requ
 		return reconcile.Result{}, nil
 	}
 
+	// Build the NamespacedName to fetch the Filesystem and make sure it exists, if not we cannot
+	// create the subvolume group
+	cephFilesystem := &cephv1.CephFilesystem{}
+	cephFilesystemNamespacedName := types.NamespacedName{Name: cephFilesystemSubVolumeGroup.Spec.FilesystemName, Namespace: request.Namespace}
+
+	if csi.EnableCSIOperator() || cephCluster.Spec.External.Enable {
+		err = csi.CreateUpdateClientProfileSubVolumeGroup(r.clusterInfo.Context, r.client, r.clusterInfo, cephFilesystemNamespacedName, buildClusterID(cephFilesystemSubVolumeGroup), cephCluster.Name)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to create ceph csi-op config CR for subVolGrp ns")
+		}
+	}
+
 	if cephCluster.Spec.External.Enable {
 		logger.Debug("external subvolume group creation is not supported, create it manually, the controller will assume it's there")
```

Comment:
This log statement indicates that the external subvolume group creation is not supported. This seems contradictory since now we add it for external mode.

Reply:
removed the condition